### PR TITLE
AFL++ redqueen update

### PR DESCRIPTION
**Pull request by tokatoka**
update with a working example & various fixes
todo:
- [x] add tokens to state in the mutator

### CODE REVIEW

File: fuzzers/fuzzbench_forkserver/src/main.rs [hunk 1]
```diff
@@ -346,7 +346,7 @@ fn fuzz(
         cmplog_shmem.write_to_env("__AFL_CMPLOG_SHM_ID").unwrap();
         let cmpmap = unsafe { cmplog_shmem.as_object_mut::<AFLppCmpMap>() };
 
-        let cmplog_observer = StdCmpObserver::new("cmplog", cmpmap, true);
+        let cmplog_observer = ForkserverCmpObserver::new("cmplog", cmpmap, true);
```
Comment (domenukk) [inline fuzzers/fuzzbench_forkserver/src/main.rs:349]: If we need different observers for different executors, can we add type constraints so they cannot be misused?

Comment (tokatoka) [reply fuzzers/fuzzbench_forkserver/src/main.rs:349]: for this ForkserverCmpObserver, No.

I think this is related issue to https://github.com/AFLplusplus/LibAFL/issues/23.
I think we have too many *hidden* dependency
"module this depends on module that" but that is not very clear if the user understands well about each module.

**Comment by s1341** [comment]
what's the status of this?

**Comment by tokatoka** [comment]
it works but it tries too many cmplog attempts, thus making it super slow..

**Pull request merged**